stubs/redis/redis/asyncio/lock.pyi

```diff
@@ -0,0 +1,44 @@
+from typing import Any, Awaitable, NoReturn
+
+from redis.asyncio import Redis
+
+class Lock:
+    lua_release: Any
+    lua_extend: Any
+    lua_reacquire: Any
+    LUA_RELEASE_SCRIPT: str
+    LUA_EXTEND_SCRIPT: str
+    LUA_REACQUIRE_SCRIPT: str
+    redis: Any
+    name: Any
+    timeout: Any
+    sleep: Any
+    blocking: Any
+    blocking_timeout: Any
+    thread_local: Any
+    local: Any
+    def __init__(
+        self,
+        redis: Redis[Any],
+        name: str | bytes | memoryview,
+        timeout: float | None = ...,
+        sleep: float = ...,
+        blocking: bool = ...,
+        blocking_timeout: float | None = ...,
+        thread_local: bool = ...,
+    ) -> None: ...
+    def register_scripts(self) -> None: ...
+    async def __aenter__(self): ...
+    async def __aexit__(self, exc_type, exc_value, traceback) -> None: ...
+    async def acquire(
+        self, blocking: bool | None = ..., blocking_timeout: float | None = ..., token: str | bytes | None = ...
+    ): ...
+    async def do_acquire(self, token: str | bytes) -> bool: ...
+    async def locked(self) -> bool: ...
+    async def owned(self) -> bool: ...
+    def release(self) -> Awaitable[NoReturn]: ...
```

Comment:
This annotation is in the source but I think it's wrong. It only NoReturns when it fails to release the lock. I'll send a PR to redis.